dms/models/dms_security_mixin.py

```diff
@@ -155,39 +164,58 @@ def _get_domain_by_inheritance(self, operation):
     @api.model
     def _get_access_groups_query(self, operation):
         """Return the query to select access groups."""
-        operation_check = {
-            "create": "AND dag.perm_inclusive_create",
-            "read": "",
-            "unlink": "AND dag.perm_inclusive_unlink",
-            "write": "AND dag.perm_inclusive_write",
-        }[operation]
-        select = f"""
-            SELECT
-                dir_group_rel.aid
-            FROM
-                dms_directory_complete_groups_rel AS dir_group_rel
-                INNER JOIN dms_access_group AS dag
-                    ON dir_group_rel.gid = dag.id
-                INNER JOIN dms_access_group_users_rel AS users
-                    ON users.gid = dag.id
-            WHERE
-                users.uid = %s {operation_check}
-            """
-        return select, (self.env.uid,)
+        # Strict validation
+        if operation not in self._OPERATION_CONDITIONS:
+            raise ValidationError(_(f"Invalid operation: {operation}"))
+
+        base_sql = """(
+                SELECT
+                    dir_group_rel.aid
+                FROM
+                    dms_directory_complete_groups_rel AS dir_group_rel
+                    INNER JOIN dms_access_group AS dag
+                        ON dir_group_rel.gid = dag.id
+                    INNER JOIN dms_access_group_users_rel AS users
+                        ON users.gid = dag.id
+                WHERE
+                    users.uid = %s
+            )"""
+        if operation == "read":
+            sql = SQL(
+                base_sql,
+                self.env.uid,
+            )
+        else:
+            base_sql = f"""(
+                            SELECT
+                                dir_group_rel.aid
+                            FROM
+                                dms_directory_complete_groups_rel AS dir_group_rel
+                                INNER JOIN dms_access_group AS dag
+                                    ON dir_group_rel.gid = dag.id
+                                INNER JOIN dms_access_group_users_rel AS users
+                                    ON users.gid = dag.id
+                            WHERE
+                                users.uid = %s {self._OPERATION_CONDITIONS[operation]}
+                        )"""
+            sql = SQL(
+                base_sql,
+                self.env.uid,
+            )
+        return sql
```

Comment:
Although the result is the same, can you reduce the diff to do everything without if/else (just as it was before)?